core/src/main/java/com/linecorp/armeria/server/DefaultUnhandledExceptionsReporter.java

```diff
@@ -59,7 +62,7 @@ final class DefaultUnhandledExceptionsReporter implements UnhandledExceptionsRep
     }
 
     @Override
-    public void report(Throwable cause) {
+    public void report(ServiceRequestContext ctx, Throwable cause) {
         if (reportingTaskFuture == null && scheduledUpdater.compareAndSet(this, 0, 1)) {
             reportingTaskFuture = CommonPools.workerGroup().next().scheduleAtFixedRate(
                     this::reportException, intervalMillis, intervalMillis, TimeUnit.MILLISECONDS);
```

Comment:
I have a question about this code. 
Why is `this::reportException` called repeatedly at regular intervals? 
I'm wondering why we don't call `this::reportException` only once when the `report` method is invoked. 
Thank you. 🙇

Reply:
Here, we lazily start the periodic timer task that logs and clears the exceptions and their count mapping every few seconds. The timer is started only once since it's guarded by the `compareAndSet()` call.

The reason we do this way is because otherwise the recently reported exceptions will not be logged until the 'next' exception is reported, which is not always the case.